packages/proxy/test/unit/http/response-middleware.spec.ts

```diff
@@ -478,6 +478,366 @@ describe('http/response-middleware', function () {
+          incomingRes: {
+            headers: {
+              'content-type': 'text/html',
+              'set-cookie': 'cookie=value',
+            },
+          },
+          req: {
+            isAUTFrame: true,
+          },
+          res: {
+            append: appendStub,
+          },
+          getRemoteState () {
+            // nonsense, but it's the simplest way to match origin policy
+            return {
+              strategy: 'file',
+              origin: 'http',
+            }
+          },
+        })
+
+        ctx.getPreviousAUTRequestUrl = () => ctx.req.proxiedUrl
+
+        await testMiddleware([CopyCookiesFromIncomingRes], ctx)
+
+        expect(appendStub).to.be.calledWith('Set-Cookie', 'cookie=value')
+      })
+
+      it('does not force SameSite=None if experimental flag is off', async function () {
+        const appendStub = sinon.stub()
+        const ctx = prepareContext({
+          incomingRes: {
+            headers: {
+              'content-type': 'text/html',
+              'set-cookie': 'cookie=value',
+            },
+          },
+          req: {
+            isAUTFrame: true,
+          },
+          res: {
+            append: appendStub,
+          },
+          config: {
+            experimentalMultiDomain: false,
+          },
+        })
+
+        await testMiddleware([CopyCookiesFromIncomingRes], ctx)
+
+        expect(appendStub).to.be.calledWith('Set-Cookie', 'cookie=value')
+      })
+
+      describe('cookie modification specifics - not Firefox', function () {
+        makeVarieties('SameSite=None; Secure', 'Secure; SameSite=None').forEach(([input, output]) => {
+          it(`${input} -> ${output}`, async function () {
+            const { appendStub, ctx } = prepareContextWithCookie(`cookie=value${input ? '; ' : ''}${input}`)
+
+            await testMiddleware([CopyCookiesFromIncomingRes], ctx)
+
+            expect(appendStub).to.be.calledOnce
+            expect(appendStub).to.be.calledWith('Set-Cookie', `cookie=value; ${output}`)
+          })
+        })
+      })
+
+      const withFirefox = {
+        getCurrentBrowser: () => ({ family: 'firefox' }),
+      }
+
+      describe('cookie modification specifics - Firefox + non-localhost', function () {
+        makeVarieties('SameSite=None; Secure', 'Secure; SameSite=None').forEach(([input, output]) => {
+          it(`${input} -> ${output}`, async function () {
+            const { appendStub, ctx } = prepareContextWithCookie(`cookie=value${input ? '; ' : ''}${input}`, {
+              req: { proxiedUrl: 'https://foobar.com' },
+              ...withFirefox,
+            })
+
+            await testMiddleware([CopyCookiesFromIncomingRes], ctx)
+
+            expect(appendStub).to.be.calledOnce
+            expect(appendStub).to.be.calledWith('Set-Cookie', `cookie=value; ${output}`)
+          })
+        })
+      })
+
+      describe('cookie modification specifics - Firefox + https://localhost', function () {
```

Comment:
nit: kinda feels like all these should be nested `describe`s in a top-level `describe('cookie modification specifics')`

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/20450/commits/c0a3ce8a5fb82b6647e2125c89488284bfd9ff6d